### ISSUE TITLE
QTableProps.rows typing shoud be Readonly to be able to use lists recieved from GraphQL

### ISSUE THREAD

**Issue by mlesin** [quasarframework/quasar#13119]
### What happened?

I'm getting rows contents to display in a table this way (using @vue/apollo-composable api):
```typescript
const users = useResult<{ id: string; name: string }[]>(result);
```
This construct returns the following type: `Readonly<ref<Readonly<{id:string;name:string}[] | undefined>>>`
But after i set this value to q-table rows property, Volar (in vscode) shows the following warning:
```
Type 'Readonly<{ id: string; name: string; }[] | undefined>' is not assignable to type 'any[] | undefined'.
  The type 'readonly { id: string; name: string; }[]' is 'readonly' and cannot be assigned to the mutable type 'any[]'.
```

### What did you expect to happen?

No type warning should appear on assignments like this.
As no possible data mutations are going to happen from inside q-table component, typing of `QTableProps.rows` property should be `Readonly<any[] | undefined>` to allow such things.

### Reproduction URL

https://codepen.io/rstoenescu/pen/xxEvBLR

### How to reproduce?

1. Reproduction link is fake because I don't find any working template of Quasar with Typescript

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Firefox, Chrome, Safari, Microsoft Edge, iOS, Android, Electron

### Quasar info output

```Shell
Operating System - Linux(5.13.0-39-generic) - linux/x64
NodeJs - 16.14.0

Global packages
  NPM - 8.3.1
  yarn - 1.22.17
  @quasar/cli - 1.3.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.6.5 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.0-beta.13 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.13.5 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.31 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.14
  pinia - 2.0.12 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.1 -- Native-ESM powered web dev build tool
  eslint - 8.12.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  @quasar/quasar-app-extension-apollo - 2.0.0-beta.3 -- A Quasar app extension to add GraphQL support using Apollo Client.

Networking
  Host - c7394f19d15a
  eth0 - 172.19.0.2
```

**Comment by mlesin** [comment]
To go further and allow better type-checking support is to introduce generic in `QTableProps` type, like the following:
```typescript
export interface QTableProps<T> {
    ...
    rows?: Readonly<T[] | undefined>;
    rowKey?: string | ((row: T) => any) | undefined;
    columns?: {
       ...
       field: string | ((row: T) => any);
       format?: (val: any, row: T) => any;
       style?: string | ((row: T) => string);
       classes?: string | ((row: T) => string);
       ...
    }
    ...
}
```
and so on, which can be easily defaulted to old behavior by supplying `QTableProps<any>`, but when exact type is supplied, it will provide much better user experience.

**Comment by Liwoj** [comment]
Oh yes, I have a very similar problem with QSelect where `options` prop is also typed as `any[] | undefined` and does not allow passing readonly array with readonly elements:

`The type 'readonly { readonly Id: string; readonly ScaleFactor: number; readonly DisplayOrder: number; }[]' is 'readonly' and cannot be assigned to the mutable type 'any[]'.`

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.6.7

**Comment by mlesin** [comment]
Thank you for fixing it so quickly!

**Comment by mityaua** [comment]
> Fix will be available in Quasar v2.6.7

Can this be considered a bug if we add the autofocus attribute? https://codepen.io/mityaua/pen/zYjQQNE?editors=1010
